core/src/test/scala/unit/kafka/server/ReplicaFetcherTierStateMachineTest.scala

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.server
+
+import kafka.cluster.BrokerEndPoint
+import org.apache.kafka.common.errors.FencedLeaderEpochException
+import org.apache.kafka.common.message.FetchResponseData
+import org.apache.kafka.common.protocol.ApiKeys
+import org.apache.kafka.common.record._
+import org.apache.kafka.common.{TopicPartition, Uuid}
+import org.junit.jupiter.api.Assertions._
+import org.junit.jupiter.api.Test
+
+import scala.collection.Map
+import scala.jdk.CollectionConverters._
+
+class ReplicaFetcherTierStateMachineTest {
+
+  val truncateOnFetch = true
+  val topicIds = Map("topic1" -> Uuid.randomUuid(), "topic2" -> Uuid.randomUuid())
+  val version = ApiKeys.FETCH.latestVersion()
+  private val failedPartitions = new FailedPartitions
+
+  private def mkBatch(baseOffset: Long, leaderEpoch: Int, records: SimpleRecord*): RecordBatch = {
```

Comment:
It seems that `mkBatch` and `initialFetchState` are not duplicated from AbstractFetcherThreadTest. Could we put them in a util class and reuse?